### DISCUSSION TITLE
AssertionError in _FanovaTree.get_marginal_variance(self, features)

### DISCUSSION THREAD

**Discussion by ttzhu813** [optuna/optuna#5295] [Q&A]
### Expected behavior

When using optuna.visualization.matplotlib.plot_param_importances(study), I expect a bar plot with importances of the parameters of the optimization problem.

### Environment

- Optuna version: 3.3
- Python version: 
- OS:
- (Optional) Other libraries and their versions:

### Error messages, stack traces, or logs

```shell
Hi there, I met the same problem with https://github.com/optuna/optuna/issues/4772, in which the stack trace was already provided.
```

### Steps to reproduce

1.
2.
3.
```python
study_file = f"test.pkl"
with open(study_file, 'rb') as f:
    study = pickle.load(f)
    storage = study._storage
    optuna.visualization.plot_param_importances(study).show()
```

### Additional context (optional)

    120 evaluator = self._evaluator
    121 evaluator.fit(
    122     X=trans_params,
    123     y=target_values,
    124     search_spaces=trans.bounds,
    125     column_to_encoded_columns=trans.column_to_encoded_columns,
    126 )

Here is the value of **_trans_params_**:
[[1.  1.  2.  0.7 1.  0.7]
 [2.3 0.  2.  2.3 1.5 0.6]
 [0.6 2.3 1.  0.3 2.1 2.9]
 [0.  1.5 2.  1.9 2.2 0.9]
 [2.8 2.2 2.  0.5 1.1 2. ]
 [1.3 1.3 2.  1.6 2.  1.8]
 [2.4 1.6 2.  1.  0.2 0.9]
 [0.3 2.  1.  1.9 1.6 2. ]
 [0.6 2.6 1.  2.3 0.9 2.7]
 [1.  0.5 1.  0.3 2.  0.4]
 [1.1 1.7 1.  0.6 2.5 2.6]
 [0.6 2.9 1.  0.6 2.2 2.3]
 [1.7 2.5 1.  1.6 2.8 2.2]
 [1.1 2.4 1.  0.2 0.8 2.8]
 [0.4 1.6 1.  0.4 2.9 2.2]
 [0.8 1.9 1.  0.7 0.6 2.7]
 [1.6 2.7 1.  0.6 0.4 2.9]
 [0.5 1.4 1.  0.8 0.8 2.4]
 [0.  1.2 1.  0.1 0.8 2.5]
 [0.5 1.5 2.  0.8 0.4 2.6]
 [0.4 1.4 1.  0.4 0.7 1.7]
 [0.9 0.6 1.  0.2 0.6 3. ]
 [0.6 0.3 1.  0.2 1.7 2.1]
 [0.2 1.8 2.  1.2 0.1 3. ]
 [1.3 0.4 1.  0.2 0.7 2.8]
 [0.8 0.1 1.  0.4 1.7 2.2]
 [1.7 0.2 1.  0.3 0.4 3. ]
 [1.5 0.2 1.  0.1 0.  2.1]
 [0.2 0.3 1.  1.3 0.3 2.7]
 [1.  0.1 1.  0.4 0.5 1.6]]

And here is the value of **_target_values_**

[1.03003126e+002 1.00404035e+002 9.71227846e+001 1.03976827e+002
 9.74407882e+001 9.96993396e+001 1.79769313e+308 9.72134700e+001
 9.91158742e+001 1.02104379e+002 1.00589943e+002 9.77478409e+001
 1.03875832e+002 1.14677322e+002 1.00428619e+002 9.63865585e+001
 9.74882889e+001 9.68413239e+001 9.62603455e+001 9.66408768e+001
 9.76967468e+001 9.51602707e+001 9.58193359e+001 9.84352036e+001
 9.57417831e+001 9.67286911e+001 9.71619797e+001 9.54917297e+001
 9.64224396e+001 9.70535812e+001]

**Comment by nzw0301** [comment]
Hi, could you share minimal reproducible codes with us? Please do not share the `test.pkl` because of security concern.

**Comment by ttzhu813** [comment]
I'm sorry about that, I only have the .pkl file.

**Reply by nzw0301** [reply]
Could you explain the reason to have `storage = study._storage`? Does your example work without it?

**Reply by ttzhu813** [reply]
Oh I forgot to delete this line when posted code, you can just ignore it.

**Comment by ttzhu813** [comment]
There has a piece of code in optuna/importance/_fanova/_tree.py, sometimes _get_node_value returns  _**inf**_, can any experts explain what does the node value represent?

`value = self._get_node_value(node_index)`

**Comment by nzw0301** [comment]
Since the code is not reproducible, we can hardly investigate this error. Let me transfer it to the discussion. I'll open an issue if this is a bug.
